superset/utils/core.py

```diff
@@ -849,6 +850,24 @@ def ping_connection(connection: Connection, branch: bool) -> None:
             # restore 'close with result'
             connection.should_close_with_result = save_should_close_with_result
 
+        if some_engine.dialect.name == "sqlite":
+
+            @event.listens_for(some_engine, "connect")
+            def set_sqlite_pragma(  # pylint: disable=unused-argument
```

Comment:
Hopefully (if approved) this will remove the need for [this](https://github.com/search?q=repo%3Aapache%2Fsuperset+backend%28%29+%3D%3D+%22sqlite%22&type=code) logic in a number of tests.